### PR TITLE
Add Misc. Hints setting

### PR DESCRIPTION
**Pull request by fenhl**
This adds a setting which when disabled removes all hints that aren't controlled by other settings:

* The Temple of Time altar (both as child and as adult) no longer tells you about the dungeon rewards, even with enhance maps/compasses off.
* The Temple of Time altar as adult no longer tells you the win conditions.
* Ganondorf no longer tells you where the light arrows are. (This allows Gossip Stones to hint about light arrows, like they do when Ganondorf access is locked behind light arrows.)

I added this setting because I think having it (and other hints) off could make random settings pictionary spoiler log runs more interesting. The setting is on by default except in Hell Mode.

Submitting as a draft because the pause menu dungeon info doesn't respect this setting yet.

### CODE REVIEW

**Comment by fenhl** [comment]
I'm having trouble getting the build script to work, could someone please check if this change to the C code works? This PR is ready to merge if it does.

**Comment by fenhl** [comment]
I managed to get it to compile but the game hangs on startup.

**Comment by fenhl** [comment]
Using an uncompressed rom fixed that, can confirm the pause menu works as intended with misc_hints off now.

**Review by Zannick — APPROVED** [review]

File: EntranceShuffle.py [hunk 1]
```diff
@@ -624,7 +624,7 @@ def validate_world(world, worlds, entrance_placed, locations_to_ensure_reachable
                 if not max_search.visited(location):
                     raise EntranceShuffleError('%s is unreachable' % location.name)
 
-    if world.shuffle_interior_entrances and \
+    if world.shuffle_interior_entrances and (world.misc_hints or world.hints != 'none') and \
```
Comment (Roman971) [inline EntranceShuffle.py:627]: Imo, this change is unrelated to the Misc Hints setting and should be kept for a separate PR. Currently, the entrance conditions here are unconditional to hint settings, so if we follow the current rules, it should remain that way regardless of the Misc. Hints setting.

In theory, we could indeed remove these special rules if there are no hints, but I think it requires discussion, and most people who would want to discuss it aren't going to expect it to come from this set of changes.

**Pull request merged**